pkg/aks/components/CruAks.vue

```diff
@@ -0,0 +1,1038 @@
+
+          const parsedError = parseAzureError(err.error || '');
+
+          this.errors.push(this.t('aks.errors.kubernetesVersions', { e: parsedError || err }));
+        });
+    },
+
+    getVmSizes() {
+      this.loadingVmSizes = true;
+      this.vmSizeOptions = [];
+      const { azureCredentialSecret, resourceLocation } = this.config;
+
+      getAKSOptions(this, azureCredentialSecret, resourceLocation, this.clusterId, 'aksVMSizes')
+        .then((res: string[]) => {
+          if (isArray(res)) {
+            this.vmSizeOptions = res.sort();
+
+            if (!this.vmSizeOptions.includes(this.defaultVmSize)) {
+              this.defaultVmSize = '';
+            }
+          }
+
+          this.loadingVmSizes = false;
+        }).catch((err: any) => {
+          this.loadingVmSizes = false;
+
+          const parsedError = parseAzureError(err.error || '');
+
+          this.errors.push(this.t('aks.errors.vmSizes', { e: parsedError || err }));
+        });
+    },
+
+    getVirtualNetworks() {
+      this.loadingVirtualNetworks = true;
+      this.virtualNetworkOptions = [];
+      const { azureCredentialSecret, resourceLocation } = this.config;
+
+      getAKSOptions(this, azureCredentialSecret, resourceLocation, this.clusterId, 'aksVirtualNetworks')
+        .then((res: any) => {
+          if (res && isArray(res)) {
+            this.virtualNetworkOptions = res;
+          }
+
+          this.loadingVirtualNetworks = false;
+        }).catch((err: any) => {
+          const parsedError = parseAzureError(err.error || '');
+
+          this.loadingVirtualNetworks = false;
+          this.errors.push(this.t('aks.errors.virtualNetworks', { e: parsedError || err }));
+        });
+    },
+
+    addPool() {
+      let poolName = `pool${ this.nodePools.length }`;
+      let mode: AKSPoolMode = 'User';
+      const _id = randomStr();
+
+      if (!this.nodePools.length) {
+        poolName = 'agentPool';
+        mode = 'System' as AKSPoolMode;
+      }
+
+      this.nodePools.push({
+        ...defaultNodePool, name: poolName, _id, mode, vmSize: this.defaultVmSize
+      });
+
+      this.$nextTick(() => {
+        if ( this.$refs.pools?.select ) {
+          this.$refs.pools.select(poolName);
+        }
+      });
+    },
+
+    removePool(idx: number) {
+      const pool = this.nodePools[idx];
+
+      removeObject(this.nodePools, pool);
+    },
+
+    selectNetwork(network: any) {
+      this.$set(this.config, 'virtualNetwork', network.name);
+      this.$set(this.config, 'virtualNetworkResourceGroup', network.resourceGroup);
+    },
+
+    setClusterName(name: string) {
+      this.$set(this.normanCluster, 'name', name);
+      this.$set(this.config, 'clusterName', name);
+    },
+
+    onMembershipUpdate(update: any) {
+      this.$set(this, 'membershipUpdate', update);
+    },
+
+    async saveRoleBindings() {
+      if (this.membershipUpdate.save) {
+        await this.membershipUpdate.save(this.normanCluster.id);
+      }
+    },
```

Comment:
The rolebinding save flow is pretty much copied from rke2.vue - in the future we should consider refactoring ClusterMembershipEditor.vue to make it easier to implement in other cluster provisioning extensions but that seems outside the scope of this PR